tests/test_interface.py

```diff
@@ -0,0 +1,138 @@
+                              msg="Method move() was not return a bool object.")
+
+    def test_03__check_if_method_correctly_is_in_area__return_bool(self):
+        self.assertIsInstance(self.r_member.is_in_area(self.r_member2.x,
+                                                       self.r_member2.y,
+                                                       self.r_member2.infection_distance), bool,
+                              msg="Method is_in_area() was not return "
+                                  "a bool object.")
+
+    def test_04__check_if_method_returns_new_instance__return_bool(self):
+        self.assertIsInstance(self.r_member.infect(), bool,
+                              msg="Method infect() was not return bool.")
+
+    def test_05__check_if_instance_was_recovered__return_bool(self):
+        self.assertIsInstance(self.r_member.recover(), bool,
+                              msg="Method recover() was not return bool.")
+
+    def test_06__check_if_instance_died__return_bool(self):
+        self.assertIsInstance(self.r_member.death(), bool,
+                              msg="Method recover() was not return bool.")
+
+
+class DGroupTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.d_member = groups.DGroup(5, 5, "member1", 0.25, 0.3, 0.2, 0.2)
+        self.d_member2 = groups.DGroup(4, 2, "member2", 0.25, 0.3, 0.2, 0.2)
+
+    def test_01__check_if_constructor_work_correctly__return_SGroup_instance(self):
+        self.assertIsInstance(self.d_member, groups.DGroup,
+                              msg="Constructor was not return a SGroup instance.")
+
+    def test_02__check_if_change_instance_position_in_grid_system__return_bool(self):
+        self.assertIsInstance(self.d_member.move(1, 2, 10, 10), bool,
+                              msg="Method move() was not return a bool object.")
+
+    def test_03__check_if_method_correctly_is_in_area__return_bool(self):
+        self.assertIsInstance(self.d_member.is_in_area(self.d_member2.x,
+                                                       self.d_member2.y,
+                                                       self.d_member2.infection_distance), bool,
+                              msg="Method is_in_area() was not return "
+                                  "a bool object.")
+
+    def test_04__check_if_method_returns_new_instance__return_bool(self):
+        self.assertIsInstance(self.d_member.infect(), bool,
+                              msg="Method infect() was not return bool.")
+
+    def test_05__check_if_instance_was_recovered__return_bool(self):
+        self.assertIsInstance(self.d_member.recover(), bool,
+                              msg="Method recover() was not return bool.")
+
+    def test_06__check_if_instance_died__return_bool(self):
+        self.assertIsInstance(self.d_member.death(), bool,
+                              msg="Method recover() was not return bool.")
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Ok, good to have some unit tests, but do you think this is enough? Maybe we need to test a functionality? eg. what if ...
Maybe it will be better to write some end to end tests with full classes behavior? So lets define couple of use cases and check if classes will behave correctly, not only in the meaning of return type, but also in the meaning of correct coordinates update, infection changes etc.